packages/broker/src/plugins/operator/fetchRedundancyFactor.ts

```diff
@@ -0,0 +1,48 @@
+import { Logger } from '@streamr/utils'
+import { Contract } from 'ethers'
+import { Operator, operatorABI } from '@streamr/network-contracts'
+import { OperatorServiceConfig } from './OperatorPlugin'
+import { z } from 'zod'
+
+const logger = new Logger(module)
+
+const MetadataSchema = z.object({
+    redundancyFactor: z.number()
+        .int()
+        .gte(1)
+})
+
+export class RedundancyFactorParseError extends Error {}
+
+export async function fetchRedundancyFactor({
+    operatorContractAddress,
+    signer
+}: Pick<OperatorServiceConfig, 'operatorContractAddress' | 'signer'>): Promise<number> {
+    const operator = new Contract(operatorContractAddress, operatorABI, signer) as unknown as Operator
+    const metadataAsString = await operator.metadata()
+
+    if (metadataAsString.length === 0) {
+        return 1
+    }
+
+    let metadata: Record<string, unknown>
+    try {
+        metadata = JSON.parse(metadataAsString)
+    } catch {
+        logger.warn('Encountered malformed metadata', { operatorContractAddress, metadataAsString })
```

Comment:
Why we both log and throw? Maybe the caller should catch the error, and logging info about the error is just one of the actions the caller does in that case.

Reply:
I want a log message with details (i.e. { operatorContractAddress, metadataAsString }) and a proper description. Doing so at a higher level would require the caller to know about details that aren't their responsibility.

One option could be to have a special error type for this?
```ts
class ErrorWithLogMessage extends Error {
    private readonly message: string
    private readonly context: Record<string, unknown>
    private readonly logLevel: LogLevel
    
    construtor(message: string, context: Record<string, unknown>, logLevel: LogLevel) {
        // set values
    }
}
```